test/test.sh

```diff
@@ -213,6 +225,38 @@ diff mpfile mpfile.get
 failures=$(($failures + (($? == 0) ? 0 : 1)))
 rm -f mpfile mpfile.get
 
+rm -f lifecycleconf lifecycleconf_new
```

Comment:
As far as I can tell, these added tests are the only thing I would be likely to take out of your pull request, unless you have some justifications for the other changes.